File: Inflearn_String/8_유효한 팰린드롬/string8.java
import java.util.*;

class Main{
  public String solution(String str){
    // String answer="NO";

    // String[] s = str.split(" ");
    // String sen="";
    // for(String x : s){
    //     for(char c : x.toCharArray()){
    //         if(Character.isAlphabetic(c)){
    //             sen+=c;
    //         }
    //     }
    // }

    // String tmp=new StringBuilder(sen).reverse().toString();
    // if(sen.equalsIgnoreCase(tmp)){
    //     answer="YES";
    // }
    // return answer;

    // GPT 코드
    int left = 0, right = str.length() - 1;
    
    while (left < right) {
      char leftChar = str.charAt(left);
      char rightChar = str.charAt(right);
      
      // 알파벳이 아닌 문자는 건너뛰기
      if (!Character.isLetter(leftChar)) {
        left++;
      } else if (!Character.isLetter(rightChar)) {
        right--;
      } else {
        // 대소문자 구분없이 비교
        if (Character.toLowerCase(leftChar) != Character.toLowerCase(rightChar)) {
          return "NO";
        }
        left++;
        right--;
      }
    }
    
    return "YES";
  }

  public static void main(String[] args){
    Main T = new Main();
    Scanner sc = new Scanner(System.in);

    String str = sc.nextLine();

    System.out.println(T.solution(str));

  }
}

File: Inflearn_String/8_유효한 팰린드롬/string8.py
# import sys

# input=sys.stdin.readline

# str=input().strip().split()

# answer=""
# for i in str:
#   for j in i:
#     if 'a'<=j<='z' or 'A'<=j<='Z':
#       answer+=j
# answer=answer.upper()

# if answer==answer[::-1]:
#   print("YES")
# else:
#   print("NO")

import sys

input=sys.stdin.readline

str=input().strip()

ans=""

left=0
right=len(str)-1

while(left<right):
  while left<right and not str[left].isalpha():
    left+=1
  while left<right and not str[right].isalpha():
    right-=1
  if left<right:
    if str[left].lower() != str[right].lower():
      ans="NO"
    left+=1
    right-=1
  ans="YES"
print(ans)